Development/nmos/control_protocol_state.cpp

```diff
@@ -0,0 +1,360 @@
+#include "nmos/control_protocol_state.h"
+
+#include "nmos/control_protocol_methods.h"
+#include "nmos/control_protocol_resource.h"
+
+namespace nmos
+{
+    namespace experimental
+    {
+        namespace details
+        {
+            // create control class
+            //   where
+            //     properties: vector of NcPropertyDescriptor can be constructed using make_control_class_property
+            //     methods:    vector of NcMethodDescriptor vs assoicated method handler where NcMethodDescriptor can be constructed using make_nc_method_descriptor
+            //     events:     vector of NcEventDescriptor can be constructed using make_nc_event_descriptor
```

Comment:
Similar ambiguity to above I think.
```suggestion
            //     events:     vector of NcEventDescriptor where NcEventDescriptor can be constructed using make_nc_event_descriptor
```